projects/schema-form/src/lib/model/ISchema.ts

```diff
@@ -37,7 +38,20 @@ export interface IProperties {
   [prop: string]: ISchema;
 }
 
-export type TSchemaPropertyType = FieldType | TNullableFieldType | 'string' | 'object' | 'array' | 'boolean' | 'integer' | 'number';
+export type TSchemaScalarPrimitiveType = 'string' | 'boolean' | 'integer' | 'number';
+export type TSchemaComplexPrimitiveType = 'object' | 'array';
+export type TSchemaNullPrimitiveType = 'null';
+export type TSchemaPrimitiveType = TSchemaComplexPrimitiveType | TSchemaScalarPrimitiveType | TSchemaNullPrimitiveType;
+
+export type TSchemaNullableScalarPrimitiveType = TUnorderedPair<
+  TSchemaNullPrimitiveType,
+  TSchemaScalarPrimitiveType
+>;
+
+export type TSchemaPropertyType = FieldType
+  | TNullableFieldType
+  | TSchemaPrimitiveType
+  | TSchemaNullableScalarPrimitiveType;
```

Comment:
hm? isn't `FieldType | TNullableFieldType` enough? `TSchemaPrimitiveType` is just a copy of `FieldType` enum values? why do we need both enum and strings at all?

Reply:
Library has another variant of working with json-schema over template syntax, and there uses FieldType enum. I don't change the logic inside this type, just a little bit cleaned up